src/lseq/lseq.rs

```diff
@@ -239,6 +251,10 @@ impl<V: Ord + Clone + Display, A: Actor + Display> LSeq<V, A> {
         println!("INCOMING CLOCK: {}", clock);
         match siblings_for_insert.get(&new_number) {
             Some(Atom::Node { payload, children }) => {
+                // (drusu) this feels weird, if we are generating the number here, we should be able to avoid these collisions.
+                //         I think we really need to be generating the full identifier when we generate the Op. Since that's what
+                //         would cause conflicts.
+
```

Comment:
I don't think having the Identifier being generated by clients change anything in the logic, the concurrent events will be happening anyway, and Identifiers could collide also even if they are not concurrent event, e.g. let's say I am as a client up to date with clocks, I decide to insert between p and q and happen to find some number between them which was already used by another Identifier, I need to find a new one, or you can just treat them as concurrent events (storing them as mini-nodes) even they are not...I guess?